src/main/scala/akka/io/AsyncDnsResolver.scala

```diff
@@ -133,11 +161,12 @@ class AsyncDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor with
 
           cache.put(Dns.Resolved(name, ipv4, ipv6), ttl)
           client ! Dns.Resolved(name, ipv4, ipv6)
-          requests -= baseId
-        case Some(CurrentRequest(client, name, None, None, Some(srv), recordsTtl)) =>
+          requests.filterNot(_._1 == baseId)
+        case Some((_, CurrentRequest(client, name, None, None, Some(srv), recordsTtl, _))) =>
           client ! SrvResolved(name, srv)
-          requests -= baseId
+          requests.filterNot(_._1 == baseId)
```

Comment:
Again, preferences live in code all of the time. This particular preference is one I've adopted for some years, and avoids introducing new abstractions. There are others: http://frozenfractal.com/blog/2016/2/19/sry-sometimes-repeat-yourself/